### PR TITLE
Update 0000-find-polymorphism.md

### PR DESCRIPTION
**Pull request by BryanCrotaz**
Add proposal for automatic hierarchy discovery

### CODE REVIEW

**Comment by xaqq** [comment]
Hello,

I never understood why this wasn't either the default behavior or available through some other means.
If anything, I don't agree with your propose to have to add a `{ polymorphic: true }`, I strongly believe this should be the default behavior.

Hopefully if this RFC is implemented we may finally have a non broken store in EmberJS !

File: text/0000-find-polymorphism.md [hunk 1]
```diff
@@ -151,5 +151,98 @@ different part of the app.
   3. This approach is incosistent with our current relationship polymorphism. It would require us to either deprecate
 `hasMany({ polymorphic: true })` (and we just released 1.0 :( ) or to use an inconsistent mixed approach.
 
+
+## Auto discovery of heirarchy
```
Comment (thomascchen) [inline text/0000-find-polymorphism.md]: I believe `heirarchy` should be spelled `hierarchy.`

File: text/0000-find-polymorphism.md [hunk 2]
```diff
@@ -151,5 +151,98 @@ different part of the app.
...
+
+The above would be much improved with auto discovery of heirarchy. This would remove the need for `abstractBaseClass` and the `polymorphic` flag on `find` etc could be inferred from whether a class has any descendants.
```
Comment (thomascchen) [inline text/0000-find-polymorphism.md]: `heirachy` -> `hierarchy` here too 🙂

**Comment by sandstrom** [comment]
Polymorphism for queries would be very useful!

Random thoughts:

- How about an adapter-level property `assumePolymorphism: true/false`. That way, some people can go with the more verbose `find('my-query', { polymorphic: true })` and others can have that as the default behaviour.
- With Ember Engines, can one assume that all models are loaded? If all models aren't loaded, then one cannot reliably determine the model-hierarchy, so auto-discovery wouldn't work. Not sure if this would be a problem in practice or not, but could be worth keeping in mind.

**Comment by wagenet** [comment]
@runspired is this something you want to pursue?

**Comment by runspired** [comment]
I don't think we need this anymore. Requests and the cache are now agnostic to type, and the identifiers RFC allows users to configure multiple types being considered as part of one identity.

**Pull request closed without merging**